src/GovUk.Education.ExploreEducationStatistics.Data.Model/Services/AbstractRepository.cs

```diff
@@ -68,7 +68,19 @@ public async Task<TEntity> FindAsync(TKey id)
 
         public IQueryable<TEntity> Find(TKey[] ids)
         {
-            return DbSet().FindAll(_context, ids.Cast<object>().ToArray());
+            var idField = typeof(TEntity).GetProperty("Id");
+            
+            var list = ids.ToList();
+            // build lambda expression
+            var parameter = Expression.Parameter(typeof(TEntity), "e");
+            var methodInfo = typeof(List<TKey>).GetMethod("Contains");
+            var body = Expression.Call(Expression.Constant(list, typeof(List<TKey>)), methodInfo, Expression.MakeMemberAccess(parameter, idField));
+            var predicateExpression = Expression.Lambda<Func<TEntity, bool>>(body, parameter);
+
+            // run query
+            return DbSet().Where(predicateExpression);
+            
+//            return DbSet().FindAll(_context, ids.Cast<object>().ToArray());
```

Comment:
I think this comment should be replaced with a TODO that mentions EES-711.

Reply:
Added TODO to the code and the original code to the ticket